src/main.cpp

```diff
@@ -17,7 +17,8 @@
 #include <znc/znc.h>
 #include <signal.h>
 
-#if defined(HAVE_LIBSSL) && defined(HAVE_PTHREAD)
+#if defined(HAVE_LIBSSL) && defined(HAVE_PTHREAD) && \
+( !defined(OPENSSL_VERSION_NUMBER) || OPENSSL_VERSION_NUMBER < 0x10100004 ) /* 1.1.0-pre4: openssl/openssl@2e52e7df518d80188c865ea3f7bb3526d14b0c08 */
```

Comment:
To quote the commit message that I pointed to in the comment about 1.1.0-pre4: "There is now no longer a need to set locking callbacks!!". The `CRYPTO_set_locking_callback` and `CRYPTO_THREADID_set...` APIs mentioned in the FAQ were all replaced with no-ops in that commit. Therefore, as far as I understand it, 1.1 now doesn't need any of this stuff and perhaps just the FAQ hasn't been updated after the recent release.